src/sparsezoo/v2/objects/model_directory.py

```diff
@@ -87,80 +83,76 @@ def __init__(
                 directory_class=NumpyDirectory,
                 display_name="sample_outputs",
                 allow_multiple_outputs=True,
+                regex=True,
             )
-        )  # key by engine name.
+        )
 
         self.sample_labels: Directory = self._directory_from_files(
             files, directory_class=Directory, display_name="sample_labels"
         )
 
         self.deployment: Directory = self._directory_from_files(
-            files, display_name="deployment", regex=False
-        )  # deployment_folder
+            files, display_name="deployment"
+        )
 
         self.onnx_folder: Directory = self._directory_from_files(
-            files, display_name="onnx", regex=False
-        )  # onnx folder
+            files,
+            display_name="onnx",
+        )
 
-        self.logs: Directory = self._directory_from_files(
-            files, display_name="logs", regex=False
-        )  # logs folder
+        self.logs: Directory = self._directory_from_files(files, display_name="logs")
 
-        self.onnx_model: File = self._file_from_files(
-            files, display_name="model.onnx"
-        )  # model.onnx
+        self.onnx_model: File = self._file_from_files(files, display_name="model.onnx")
 
-        self.analysis: File = self._file_from_files(
-            files, display_name="analysis.yaml"
-        )  # analysis.yaml
+        self.analysis: File = self._file_from_files(files, display_name="analysis.yaml")
         self.benchmarks: File = self._file_from_files(
             files, display_name="benchmarks.yaml"
-        )  # benchmarks.yaml
-        self.eval_results: File = self._file_from_files(
-            files, display_name="eval.yaml"
-        )  # eval.yaml
+        )
+        self.eval_results: File = self._file_from_files(files, display_name="eval.yaml")
 
         self.recipes: List[File] = self._file_from_files(
             files, display_name="recipe(.*).md", regex=True
-        )  # recipe{_tag}.md
+        )
 
         # sorting name of `sample_inputs` and `sample_output` files,
-        # so that they have same correspondence when we jointly
+        # so that they have same one-to-one correspondence when we jointly
         # iterate over them
         self.sample_inputs.files.sort(key=lambda x: x.name)
         [
             sample_outputs.files.sort(key=lambda x: x.name)
             for sample_outputs in self.sample_outputs.values()
         ]
 
-        files = [
-            self.training,
-            self.sample_originals,
-            self.sample_inputs,
-            self.sample_outputs,
-            self.sample_labels,
-            self.deployment,
-            self.logs,
-            self.onnx_folder,
-            self.onnx_model,
-            self.analysis,
-            self.benchmarks,
-            self.eval_results,
-            self.model_card,
-            self.recipes,
-        ]
+        self._files_dictionary = {
```

Comment:
What's the reasoning behind the change from list to dictionary? Is the `key` here better than using  file/directory `.name`?

Reply:
Yeah, this is to easy iterate over files for download in `L242` (have name for messages/errors together with the file).